python/ray/internal/internal_api.py

```diff
@@ -220,3 +223,157 @@ def free(object_refs, local_only=False):
+    import requests
+
+    if node_id and node_ip:
+        raise ValueError(
+            f"Node id {node_id} and node ip {node_ip} are given at the same time. Please provide only one of them."
+        )
+    gcs_addr = services.get_ray_address_from_environment()
+    url = _get_dashboard_url()
+    nodes = []
+    if node_id:
+        nodes.append(
+            json.loads(requests.get(f"{url}/nodes/{node_id}").text)["data"]["detail"]
+        )
+    else:
+        nodes = json.loads(requests.get(f"{url}/nodes?view=details").text)["data"][
+            "clients"
+        ]
+    parsed_node_info = []
+    state = GlobalState()
+    if use_gcs_for_bootstrap():
+        options = GcsClientOptions.from_gcs_address(gcs_addr)
+    else:
+        options = GcsClientOptions.from_redis_address(
+            gcs_addr, ray_constants.REDIS_DEFAULT_PASSWORD
+        )
+    state._initialize_global_state(options)
+    resources = state._available_resources_per_node()
+    for node in nodes:
+        if node["raylet"] != {}:
+            info = {
+                "id": node["raylet"]["nodeId"],
+                "ip": node["ip"],
+                "state": node["raylet"]["state"],
+                "available_resources": resources[node["raylet"]["nodeId"]],
+                "top_10_process_mem_usage": node["top10ProcMemUsage"],
+            }
+            if debug:
+                info["logUrl"] = node["logUrl"]
+            parsed_node_info.append(info)
+    results = []
+    # Filtering.
+    if node_ip:
+        for node in parsed_node_info:
+            if node["ip"] == node_ip:
+                results.append(node)
+    else:
+        results = parsed_node_info
+    return results
+
+
+def ray_actors(actor_id: str):
+    import requests
+
+    url = _get_dashboard_url()
+    result = json.loads(requests.get(f"{url}/logical/actors").text)["data"]["actors"]
+    if actor_id:
+        result = {actor_id: result[actor_id]}
+    # Parsing.
+    output = {}
+    for actor_id, data in result.items():
+        func_desc = data["taskSpec"]["functionDescriptor"]
+        actor_info = {}
+        actor_info["class_name"] = func_desc[next(iter(func_desc))]["className"]
+        actor_info["actor_id"] = data["actorId"]
+        actor_info["node_id"] = data["address"]["rayletId"]
+        actor_info["job_id"] = data["jobId"]
+        actor_info["name"] = data["name"]
+        # actor_info["namespace"] = data["rayNamespace"]
+        actor_info["pid"] = data["pid"]
+        actor_info["ip"] = data["address"]["ipAddress"]
+        # actor_info["resource_req"] = data["resourceMapping"]
+        actor_info["state"] = data["state"]
+        output[actor_id] = actor_info
+    return output
+
+
+def ray_log(
+    ip_address: str,
+    node_id: str,
+    filters: List[str],
+    limit: int = 100,
+):
+    """Return the `limit` number of lines of logs."""
+    import requests
+
+    dashboard_url = _get_dashboard_url()
+    nodes = ray_nodes(node_id, ip_address, debug=True)
```

Comment:
In the real implementation, we might want to just have a single API, and the server can find the IP address instead of doing it in the client.